### PR TITLE
Introduce JS testing

### PR DESCRIPTION
**Pull request by stevenwilkin**
As mentioned in #20, the JS app has some upcoming work which will see it use a lot more logic than before when it comes to dealing with the locations data so let's get it under test.

This PR includes a refactor which extracts the code used to generate the content displayed for each location on the map into a separate function and places a few tests around it.

### CODE REVIEW

**Comment by gjvis** [comment]
:metal:

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'jasmine'
+
+load 'jasmine/tasks/jasmine.rake'
+
+ENV['JASMINE_CONFIG_PATH'] = 'spec/support/jasmine.yml'
```
Comment (andrewgarner) [inline Rakefile]: Should this be `ENV['JASMINE_CONFIG_PATH'] ||= 'spec/support/jasmine.yml'` so that it can be configured elsewhere if required?

Comment (stevenwilkin) [inline Rakefile]: It was that originally but I figured it would never be set anywhere else

Comment (andrewgarner) [inline Rakefile]: Never say never.

**Comment by stevenwilkin** [comment]
Updated per @andrewgarner comment

File: Rakefile (hunk 1 above)
Comment (gjvis) [inline Rakefile]: :+1:

**Pull request merged**